File: VB/09_Interaction/SpecifyPageToView.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Pdf.Actions
Imports Spire.Pdf.General
Imports Spire.Pdf
Namespace SpecifyPageToView
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'pdf file 
			Dim input As String = "..\..\..\..\..\..\Data\Sample3.pdf"

			'open pdf document
			Dim doc As New PdfDocument(input)

			'get the fourth page
			Dim page As PdfPageBase = doc.Pages(3)

			'create a destination
			Dim destination As New PdfDestination(page)

			'create GoToAction instance
			Dim action As New PdfGoToAction(destination)

			'set the action to execute when the document is opened.
			doc.AfterOpenAction = action

			Dim output As String = "SpecifyPageToView.pdf"

			'save pdf document
			doc.SaveToFile(output)

			'Launching the Pdf file
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Interaction/SpecifyPageToView.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf.Actions;
using Spire.Pdf.General;
using Spire.Pdf;
namespace SpecifyPageToView
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //pdf file 
            string input = "..\\..\\..\\..\\..\\..\\Data\\Sample3.pdf";

            //open pdf document
            PdfDocument doc = new PdfDocument(input);

            //get the fourth page
            PdfPageBase page = doc.Pages[3];

            //create a destination
            PdfDestination destination = new PdfDestination(page);

            //create GoToAction instance
            PdfGoToAction action = new PdfGoToAction(destination);

            //set the action to execute when the document is opened.
            doc.AfterOpenAction = action;

            string output = "SpecifyPageToView.pdf";

            //save pdf document
            doc.SaveToFile(output);

            //Launching the Pdf file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
